planner/core/planbuilder.go

```diff
@@ -72,25 +77,29 @@ func (info *tableHintInfo) ifPreferINLJ(tableNames ...*model.CIStr) bool {
 // Which it joins on with depend on sequence of traverse
 // and without reorder, user might adjust themselves.
 // This is similar to MySQL hints.
-func (info *tableHintInfo) matchTableName(tables []*model.CIStr, tablesInHints []model.CIStr) bool {
+func (info *tableHintInfo) matchTableName(tables []*model.CIStr, joinHint *joinHintInfo) bool {
 	for _, tableName := range tables {
 		if tableName == nil {
 			continue
 		}
-		for _, curEntry := range tablesInHints {
+		for _, curEntry := range joinHint.tables {
 			if curEntry.L == tableName.L {
+				joinHint.matched = true
```

Comment:
Do we need to mark hint as unmatched if any specified hint table cannot find a match? /cc @morgo

Reply:
The default value is `false`, is unmatched?

Reply:
I mean if the hint contains `t1` and `t2`, while the query only involves `t2`, should we raise warning?

Reply:
If so, maybe it's more reasonable to raise a warning.

Reply:
Can we define the matched as a string slice?
Thus we can print the accurate wrong table name in the warning message

Reply:
Warning sounds good to me.  Note that in some cases, MySQL hints do not even raise warnings because they are considered hints (entirely optional to enforce vs. directives).  I think we should just arrive at warnings and never errors.